File: Computer-Introduction/HW1/SDNN.py
import numpy

def main():
    """
    main function
    """
    with open('E64061151.2018-09-27-06-07-48.peakposition.txt','r') as f:
        text_list = f.readlines()
        data = dataProcess(text_list)
        interval = Interval(data)
        # ans = numpy.std(interval)
        mean = numpy.mean(interval)
        ans = numpy.sqrt(numpy.mean([(i - mean)**2 for i in interval]))
        print("SDNN : ",ans)

def dataProcess(text_list:list):
    """
    convert the text_list readed from peakposition.txt to integer

    parameter :
    type is str list

    return the integer list data
    """
    data = [int(i) for i in text_list]
    return data

def Interval(data:list):
    length = len(data)
    interval = []
    interval.append(2*data[0])
    for i in range(1,length):
        interval.append(2*(data[i] - data[i-1]))
    return interval

if __name__ == '__main__':
    main()

File: Computer-Introduction/HW1/SDNN.cpp
#include<math.h>
#include<vector>
#include<iostream>

using namespace std;

int main(){
    int temp;
    double mean = 0;
    double ans = 0;
    vector<int> data;
    FILE * file = NULL;
    file = fopen("E64061151.2018-09-27-06-07-48.peakposition.txt","r");
    if(file == NULL) throw "Cannot open File";
    else{
        while(EOF != fscanf(file,"%d",&temp)){
            data.push_back(temp);
        }
    }
    for(int i = 0; i < data.size() - 1; i++)
        data[i] = (data[i + 1] - data[i]) * 2;
    data.pop_back();

    for(int i = 0; i < data.size(); i++)
        mean += data[i];
    mean /= (double)data.size();

    for(int i = 0; i < data.size(); i++)
        ans += pow((data[i] - mean),2);
    ans = sqrt(ans/(double)data.size());

    printf("SDNN = %.8f\n",ans);
}
